File: summer2013/openmp/fortran/23_simple_test4.f90
program simple_test4
    integer, parameter :: N = 4
    integer :: i, tid, omp_get_thread_num

    call omp_set_num_threads(4)
!$omp parallel private(i, tid)
    tid = omp_get_thread_num()
    !$omp sections
    !$omp section
    do i=0, N-1
        print *, 'L1 tid=',tid
    end do
    !$omp section
    do i=0, N-1
        print *, 'L2 tid=',tid
    end do
    call sleep(2);
    !$omp end sections
    print *, 'end tid=', tid
!$omp end parallel
end

File: summer2013/example_openmp/c/23_simple_test4.c
#include <stdio.h>
#include <omp.h>
#define N 4

int main()
{
	int i, tid;
	omp_set_num_threads(4);
#pragma omp parallel private(i, tid)
{
	tid = omp_get_thread_num();
	#pragma omp sections
	{
		#pragma omp section
		{
			for(i=0; i<N; i++)
				printf("L1 tid=%d\n", tid);
		}
		#pragma omp section
		{
			for(i=0; i<N; i++)
				printf("L2 tid=%d\n", tid);
			sleep(2);
		}
	}
	printf("end tid=%d\n", tid);
} // pragma omp parallel
}
